smithy-build/src/main/java/software/amazon/smithy/build/model/ConfigLoader.java

```diff
@@ -115,6 +115,11 @@ public Node stringNode(StringNode node) {
             while (matcher.find()) {
                 String variable = matcher.group(1);
                 String replacement = expand(node.getSourceLocation(), variable);
+                // INLINE over-matches to allow for escaping. If the over-matched first group does not start with
+                // '${', we need to prepend the first character from that group on the replacement.
+                if (!matcher.group(0).startsWith("${")) {
```

Comment:
This works, but it would probably be easier to understand if this used https://github.com/awslabs/smithy/blob/master/smithy-utils/src/main/java/software/amazon/smithy/utils/SimpleParser.java. I think we can merge this for now to fix the bug, and then go back and improve this when we have time.